File: srm/srm-gateway/src/main/java/com/srm/gateway/entity/CooperationRecord.java
package com.srm.gateway.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("cooperation_record")
public class CooperationRecord {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long supplierId;
    
    private LocalDate startDate;
    
    private LocalDate endDate;
    
    private String cooperationType;
    
    private String contractNo;
    
    private BigDecimal amount;
    
    private String currency;
    
    private String status;
    
    private String description;
    
    private String remark;
    
    @TableField(fill = FieldFill.INSERT)
    private String createdBy;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedBy;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    
    @TableLogic
    private Integer delFlag;
}

File: srm-frontend-app/src/services/cooperationRecord.ts
import api from './api';
import type { ApiResponse } from '../types';

export interface CooperationRecord {
  id: number;
  supplierId: number;
  startDate?: string;
  endDate?: string;
  cooperationType?: string;
  contractNo?: string;
  amount?: number;
  currency?: string;
  status?: string;
  description?: string;
  remark?: string;
  createdAt?: string;
  updatedAt?: string;
  delFlag: number;
}

export interface CooperationRecordCreate {
  supplierId: number;
  startDate?: string;
  endDate?: string;
  cooperationType?: string;
  contractNo?: string;
  amount?: number;
  currency?: string;
  status?: string;
  description?: string;
  remark?: string;
}

export const cooperationRecordApi = {
  getBySupplierId: (supplierId: number): Promise<ApiResponse<CooperationRecord[]>> => {
    return api.get(`/api/cooperation-records/supplier/${supplierId}`);
  },

  getById: (id: number): Promise<ApiResponse<CooperationRecord>> => {
    return api.get(`/api/cooperation-records/${id}`);
  },

  create: (data: CooperationRecordCreate): Promise<ApiResponse<number>> => {
    return api.post('/api/cooperation-records', data);
  },

  update: (id: number, data: Partial<CooperationRecord>): Promise<ApiResponse<void>> => {
    return api.put(`/api/cooperation-records/${id}`, data);
  },

  delete: (id: number): Promise<ApiResponse<void>> => {
    return api.delete(`/api/cooperation-records/${id}`);
  },
};
